common/ledger/blkstorage/blockindex.go

```diff
@@ -288,19 +283,30 @@ func (index *blockIndex) exportUniqueTxIDs(dir string, newHashFunc snapshot.NewH
 			continue
 		}
 		previousTxID = txID
+		if numTxIDs == 0 { // first iteration, create the data file
+			dataFile, err = snapshot.CreateFile(filepath.Join(dir, snapshotDataFileName), snapshotFileFormat, newHashFunc)
+			if err != nil {
+				return nil, err
+			}
+			defer dataFile.Close()
+		}
 		if err := dataFile.EncodeString(txID); err != nil {
 			return nil, err
 		}
 		numTxIDs++
 	}
 
+	if numTxIDs == 0 {
+		return nil, nil
+	}
```

Comment:
Nit. 

Just to maintain consistency, similar to what is done in config history export, we can use `if dataFile == nil` here.

Reply:
sure